### PR TITLE
Add option to track and highlight memory changes

### PR DESCRIPTION
**Pull request by Grazfather**
Implements #167.

Note that this could be improved: I basically won't highlight if the address jumps around (e.g. using `-r`. We could figure out if the new address buf overlaps the previous (even if the start changed) and maintain our highlights, but I don't think it's worth it.

I can get this to _crash_ if I shrink the window, hit a BP, and then grow the window and hit another. This is because this causes an `IndexError` when the cache is checked beyond its old capacity.

### CODE REVIEW

File: voltron/plugins/view/memory.py [hunk 1]
```diff
@@ -85,13 +88,21 @@ def render(self, results):
...
+                                bytes.append(self.colour("%02X" % x, 'red'))
+                            else:
+                                bytes.append("%02X" % x)
+                        memory_str = ' '.join(bytes)
+                    else:
+                        memory_str = ' '.join(["%02X" % x for x in six.iterbytes(chunk)])
+
                     if self.args.deref:
-                        info_str = ''
-                        if len(chunk) == target['addr_size']:
```
Comment (Grazfather) [inline voltron/plugins/view/memory.py]: I think this check was redundant: `len(chunk)` comes from the slice above, whose length is `self.args.bytes` (see [here](https://github.com/snare/voltron/pull/168/files#diff-666f65600001e7d8fada866226fbee57R88)), but `self.args.bytes` is already set to match `target['addr_size']` [here](https://github.com/snare/voltron/pull/168/files#diff-666f65600001e7d8fada866226fbee57R83)

File: voltron/plugins/view/memory.py [hunk 2]
```diff
@@ -85,13 +88,21 @@ def render(self, results):
                     chunk = m_res.memory[c:c + self.args.bytes]
                     addr_str = self.colour(self.format_address(m_res.address + c, size=target['addr_size'], pad=False),
                                            self.config.format.addr_colour)
+
+                    if self.args.track and self.last_memory and self.last_address == m_res.address:
+                        bytes = []
+                        for i, x in enumerate(six.iterbytes(chunk)):
+                            if x != self.last_memory[c + i]:
+                                bytes.append(self.colour("%02X" % x, 'red'))
```
Comment (Grazfather) [inline voltron/plugins/view/memory.py]: Could use a configuration colour here, but the one used in the registers view for tracking changes didn't seem to work here for some reason.

**Comment by snare** [comment]
Awesome man, looks good. I'll merge this now, but if you wanna open an issue for the improvements and edge cases that need handling that'd be great

File: voltron/plugins/view/memory.py (hunk 2 above)
Comment (snare) [inline voltron/plugins/view/memory.py]: I added a new config option for the stack and memory views and stuck it in here.

**Pull request merged**